### PR TITLE
Parse and render, Model to Text and reverse.

### PR DESCRIPTION
**Pull request by k-y0u**
Parse and render, Model to Text and reverse.

### CODE REVIEW

File: src/components/menu/GitBranchMenu.vue [hunk 1]
```diff
@@ -102,7 +102,8 @@ import DialogEvent from 'src/composables/events/DialogEvent';
 const props = defineProps({
   currentBranchName: {
     type: String,
-    required: true,
+    // required: true,
```
Comment (Zorin95670) [inline src/components/menu/GitBranchMenu.vue:105]: to delete

File: src/composables/Project.js [hunk 2]
```diff
@@ -202,6 +202,28 @@ export async function getCurrentBranch(projectId) {
     fs,
     dir: `/${projectId}`,
     fullname: false,
+    test: true,
+  });
+}
+
+export async function readFsFile(projectId, fileInformation) {
```
Comment (Zorin95670) [inline src/composables/Project.js:209]: why not `readProjectFile` ?

File: src/components/ModelizerTextView.vue [hunk 3]
```diff
@@ -41,9 +42,12 @@ import {
...
-import { getProjectFiles, readProjectFile } from 'src/composables/Project';
+import { getProjectFiles, readProjectFile, writeProjectFile } from 'src/composables/Project';
 import FileEvent from 'src/composables/events/FileEvent';
 import GitEvent from 'src/composables/events/GitEvent';
+import PluginEvent from 'src/composables/events/PluginEvent';
+import { getPlugins } from 'src/composables/PluginManager';
+import { FileInformation } from 'leto-modelizer-plugin-core';
 
 const props = defineProps({
   viewType: {
```
Comment (justinehell) [inline src/components/ModelizerTextView.vue:53]: to delete if `viewType` prop not passed to `MonacoEditor.vue`

File: src/pages/ModelizerPage.vue [hunk 4]
```diff
@@ -13,7 +13,10 @@
...
         data-cy="modelizer-page"
       >
-        <modelizer-model-view v-show="viewType === 'model'" />
+        <modelizer-model-view
+          v-show="viewType === 'model'"
+          :project-name="projectName"
+        />
         <modelizer-text-view
           v-show="viewType === 'text'"
           :viewType="viewType"
```
Comment (justinehell) [inline src/pages/ModelizerPage.vue:22]: to delete if `viewType` not used inside `ModelizerTextView`

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/pages/ModelizerPage.vue (hunk 4 above)
Comment (justinehell) [inline src/pages/ModelizerPage.vue:22]: to delete `:viewType="viewType"` is props not used inside ModelizerTextView

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/components/menu/GitBranchMenu.vue (hunk 1 above)
Comment (k-y0u) [reply src/components/menu/GitBranchMenu.vue:105]: Done

File: src/components/ModelizerModelView.vue [hunk 5]
```diff
@@ -29,13 +29,44 @@ import {
...
+    .filter((fileInfo) => plugin.parser.isParsable(fileInfo))
+    .map((fileInfo) => readFsFile(props.projectName, fileInfo)))
+    .then((allResults) => allResults
+      .filter((result) => result.status === 'fulfilled')
+      .map((result) => result.value));
+}
+
+async function getFileInformations() {
+  return getProjectFiles(props.projectName);
+}
```
[the remark this answers is not in the document]
Comment (k-y0u) [reply src/components/ModelizerModelView.vue:58]: Done

File: src/components/ModelizerTextView.vue (hunk 3 above)
Comment (k-y0u) [reply src/components/ModelizerTextView.vue:53]: Done

File: src/pages/ModelizerPage.vue (hunk 4 above)
Comment (k-y0u) [reply src/pages/ModelizerPage.vue:22]: Done

File: src/components/ModelizerTextView.vue [hunk 6]
```diff
@@ -21,9 +21,10 @@
     >
       <template v-slot="{ file }">
         <monaco-editor
-          :content="file.content"
-          :viewType="viewType"
+          :fileInput="file"
+          :project-name="projectName"
         />
+        <!-- :viewType="viewType" -->
```
[the remark this answers is not in the document]
Comment (k-y0u) [reply src/components/ModelizerTextView.vue]: Done

File: src/composables/Project.js (hunk 2 above)
Comment (k-y0u) [reply src/composables/Project.js]: Deleted (duplicate)

File: src/components/menu/GitBranchMenu.vue [hunk 7]
```diff
@@ -102,7 +102,7 @@ import DialogEvent from 'src/composables/events/DialogEvent';
 const props = defineProps({
   currentBranchName: {
     type: String,
-    required: true,
+    default: '',
```
Comment (Zorin95670) [inline src/components/menu/GitBranchMenu.vue:105]: remove this modif

File: src/composables/events/FileEvent.js [hunk 8]
```diff
@@ -45,11 +45,18 @@ const DeleteFileEvent = new Subject();
  */
 const ExpandFolderEvent = new Subject();
 
+/**
+ * Represent a rxjs Event object to emit and to receive events about file update.
+ * @typedef {Subject} UpdateFileEvent
+ */
+const UpdateFileEvent = new Subject();
```
Comment (Zorin95670) [inline src/composables/events/FileEvent.js:52]: To remove not used, no next is done

Comment (k-y0u) [reply src/composables/events/FileEvent.js]: Done

File: src/components/menu/GitBranchMenu.vue (hunk 7 above)
Comment (k-y0u) [reply src/components/menu/GitBranchMenu.vue]: Done

**Review by Zorin95670 — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

**Pull request merged**